diff --git a/Makefile.am b/Makefile.am
index 7e661f60..73fb2827 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -161,7 +161,7 @@ if !WITH_ASSERTIONS
 blockalign_CXXFLAGS +=-DNDEBUG
 endif
 
-install-exec-local: defaults_client init.d_client init.d_client_rh
+disable-install-exec-local: defaults_client init.d_client init.d_client_rh
 if INSTALL_DEBIAN_INITD
 	$(INSTALL) -D "$(srcdir)/init.d_client" "$(sysconfdir)/init.d/urbackupclientbackend"
 	if ! test -e "$(sysconfdir)/default/urbackupclientbackend"; then $(INSTALL) -D "$(srcdir)/defaults_client" "$(sysconfdir)/default/urbackupclientbackend"; fi
@@ -205,7 +205,7 @@ install-data-local: urbackupclient/backup_scripts/list urbackupclient/backup_scr
 					client/version.txt client/data/urbackup_ecdsa409k1.pub client/data/updates_h.dat
 endif
 
-	$(MKDIR_P) "$(DESTDIR)$(localstatedir)/urbackup/"
+	echo WARNING: disabled running command "(MKDIR_P) \"$(DESTDIR)$(localstatedir)/urbackup/\""
 	$(MKDIR_P) "$(DESTDIR)$(datadir)/urbackup/scripts"
 	$(MKDIR_P) "$(DESTDIR)$(sysconfdir)/urbackup"
 	
@@ -231,7 +231,7 @@ endif
 	for script in "$(DESTDIR)$(datadir)/urbackup/scripts/"*; do sed "s|SYSCONFDIR|$(sysconfdir)/urbackup|g" "$$script" > "$$script.r"; mv "$$script.r" "$$script"; done
 	chmod 700 "$(DESTDIR)$(datadir)/urbackup/scripts/"*
 	chmod 700 "$(DESTDIR)$(sysconfdir)/urbackup/"*
-	$(INSTALL_DATA) "$(srcdir)/client/version.txt" "$(DESTDIR)$(localstatedir)/urbackup/version.txt"
+	echo WARNING: disabled running command "$(INSTALL_DATA) \"$(srcdir)/client/version.txt\" \"$(DESTDIR)$(localstatedir)/urbackup/version.txt\""
 	$(INSTALL_DATA) "$(srcdir)/client/data/urbackup_ecdsa409k1.pub" "$(DESTDIR)$(datadir)/urbackup/"
 	$(INSTALL_DATA) "$(srcdir)/client/data/updates_h.dat" "$(DESTDIR)$(datadir)/urbackup/"
 	